src/api/tests/products/post-api-products.spec.ts

```diff
@@ -0,0 +1,221 @@
+import { IAPICredentials } from 'types/signIn.types';
+import { test, expect } from 'fixtures/index';
+import { STATUS_CODES } from 'data/status.code';
+import { USER_LOGIN, USER_PASSWORD } from 'config/environment';
+import { generateProductData } from 'data/products/generateProduct.data';
+import { validateSchema } from 'utils/notifications/validations/schemaValidation';
+import { productSchema } from 'data/schemas/products/product.schema';
+import _ from 'lodash';
+import { baseSchema } from 'data/schemas/base.schema';
+import { faker } from '@faker-js/faker';
+
+test.describe('[API] [Products] POST /api/products', () => {
+  const user: IAPICredentials = { username: USER_LOGIN, password: USER_PASSWORD };
+  let token: string;
+  const productIds = new Set<string>();
+
+  test.beforeEach(async ({ signInController }) => {
+    const signInResponse = await signInController.signIn(user);
+    token = signInResponse.headers['authorization'];
+    expect.soft(signInResponse.status).toBe(STATUS_CODES.OK);
+  });
+
+  test.afterEach(async ({ productController }) => {
+    const array = [...productIds];
+    for (const id of array) {
+      const responseDelete = await productController.delete(id, token);
+      expect.soft(responseDelete.status).toBe(STATUS_CODES.DELETED);
+    }
+    productIds.clear();
+  });
+
+  test('Should create product with valid data', async ({ productController }) => {
+    const productBody = generateProductData();
+    const response = await productController.create(productBody, token);
+
+    validateSchema(productSchema, response.body);
+    expect.soft(productBody).toMatchObject(_.omit(response.body.Product, ['_id', 'createdOn']));
+    expect.soft(response.status).toBe(STATUS_CODES.CREATED);
+
+    const productId = response.body.Product._id;
+    productIds.add(productId);
+  });
+
+  test('Should create product with MIN valid value of attribute "amount" ', async ({
```

Comment:
Лучше явно указывать, не нужно держать в голове все валидации для каждого поля
```suggestion
  test('Should create product with 0 amount value', async ({
```